c10/core/SymInt.cpp

```diff
@@ -39,9 +39,11 @@ SymInt SymInt::operator*(SymInt sci) const {
   // TODO: technically we need to check that the classes match
   if (!a) {
     a = common->wrap(data_);
+    toSymInt(a); //
```

Comment:
debug code?

Reply:
`toSymInt` has a side-effect of persisting `SymIntNode`s into a `SymIntTable`, so a comment would probably help why we need to do it given that `SymInt::toSymInt(a->add(b));` should have strong references to both `a` and `b`?

Reply:
This code is completely broken anyway; every time you convert SymbolicIntNode to SymInt the current code allocates a fresh table entry which is wrong